### PR TITLE
Update core dependency and remove usage of deprecated APIs

### PR DESCRIPTION
**Pull request by dmitryax**
Remove usages of:
- Map.Upsert
- pcommon.InvalidTraceID
- pcommon.InvalidSpanID

### CODE REVIEW

File: exporter/prometheusremotewriteexporter/testutil_test.go [hunk 1]
```diff
@@ -345,7 +345,7 @@ func getSummaryMetric(name string, attributes pcommon.Map, ts uint64, sum float6
 	dp.SetCount(count)
 	dp.SetSum(sum)
 	attributes.Range(func(k string, v pcommon.Value) bool {
```
Comment (bogdandrutu) [inline exporter/prometheusremotewriteexporter/testutil_test.go:347]: attributes.CopyTo(dp.Attributes())? Can be a separate PR.

File: internal/coreinternal/goldendataset/span_generator.go [hunk 2]
```diff
@@ -363,16 +363,14 @@ func appendMaxCountAttributes(includeStatus bool, attrMap pcommon.Map) {
...
-	subMap.MapVal().InsertBool("flow-alt3", false)
-	attrMap.Upsert("application.abflags", subMap)
+	stages := attrMap.UpsertEmptySlice("application.stages")
+	stages.AppendEmpty().SetStringVal("Launch")
+	stages.AppendEmpty().SetStringVal("Injestion")
+	stages.AppendEmpty().SetStringVal("Validation")
+	subMap := attrMap.UpsertEmptyMap("application.abflags")
+	subMap.InsertBool("UIx", false)
+	subMap.InsertBool("UI4", true)
+	subMap.InsertBool("flow-alt3", false)
```
Comment (bogdandrutu) [inline internal/coreinternal/goldendataset/span_generator.go:373]: s/InsertBool/UpsertBool

Can be a separate PR.

File: pkg/telemetryquerylanguage/contexts/internal/tqlcommon/map.go [hunk 3]
```diff
@@ -36,5 +36,5 @@ func SetMapValue(attrs pcommon.Map, mapKey string, val interface{}) {
 	}
 
 	SetValue(value, val)
-	attrs.Upsert(mapKey, value)
+	value.CopyTo(attrs.UpsertEmpty(mapKey))
```
Comment (bogdandrutu) [inline pkg/telemetryquerylanguage/contexts/internal/tqlcommon/map.go:39]: SetValue(attrs.UpsertEmpty(mapKey), val)

Should be a separate PR.

**Review by bogdandrutu — APPROVED** [review]

File: pkg/telemetryquerylanguage/contexts/internal/tqlcommon/map.go (hunk 3 above)
Comment (dmitryax) [reply pkg/telemetryquerylanguage/contexts/internal/tqlcommon/map.go:39]: Yeah I wanted to keep this PR's scope limited and apply other improvements after
